### PR TITLE
[svgicons2svgfont]: add definition for nfroidure/svgicons2svgfont

### PR DESCRIPTION
**Pull request by casaper**
https://github.com/nfroidure/svgicons2svgfont

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

~~If changing an existing definition:~~

~~If removing a declaration:~~

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## svgicons2svgfont ([<kbd>unpkg</kbd>](https://unpkg.com/browse/svgicons2svgfont@latest/))
was missing the following properties:
1. once
1. EventEmitter
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 59a6e10d391f66dde7d045ad99a76ecd080a74ed
</p>

**Comment by casaper** [comment]
> Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
> The check for missing properties isn't always right, so take this list as advice, not a requirement.
> ## svgicons2svgfont ([unpkg](https://unpkg.com/browse/svgicons2svgfont@latest/))
> 
> was missing the following properties:
> 
>     1. once
> 
>     2. EventEmitter
> 
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [59a6e10](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/59a6e10d391f66dde7d045ad99a76ecd080a74ed)

Truly sorry, but I'm unable to decipher how I could get these included. I don't seem to understand their code thus far.

The purpose of my addition is mainly to get the options for that lib typed.

TIA

File: types/svgicons2svgfont/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+declare class SVGIcons2SVGFont extends Stream.Transform {
+    constructor(options?: SVGIcons2SVGFont.SvgIcons2FontOptions);
+}
+
+declare namespace SVGIcons2SVGFont {
+    interface SvgIcons2FontOptions {
+        /**
+         * The font family name you want
+         *
+         * Default value: 'iconfont'
```
Comment (peterblazejewicz) [inline types/svgicons2svgfont/index.d.ts:19]:
```suggestion
         * @default 'iconfont'
```

File: types/svgicons2svgfont/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+         */
+        fontName?: string;
+        /**
+         * The font id you want (Default value: the options.fontName)
+         *
+         * Default value: the options.fontName value
```
Comment (peterblazejewicz) [inline types/svgicons2svgfont/index.d.ts:25]:
```suggestion
         * @default the options.fontName value
```

File: types/svgicons2svgfont/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+         */
+        normalize?: boolean;
+        /**
+         * The outputted font height (defaults to the height of the highest input icon).
+         */
+        fontHeight?: number;
+        /**
+         * Setup SVG path rounding.
+         *
+         * Default value: 10e12
```
Comment (peterblazejewicz) [inline types/svgicons2svgfont/index.d.ts:55]:
```suggestion
         * @default 10e12
```

File: types/svgicons2svgfont/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+         *
+         * You can set any character data in but it is the be suited place for a copyright mention.
+         */
+        metadata?: string;
+        /**
+         * Allows you to provide your own logging function.
+         *
+         * Set to function(){} to disable logging.
+         *
+         * Defaults to console.log()
```
Comment (peterblazejewicz) [inline types/svgicons2svgfont/index.d.ts:84]:
```suggestion
         * @default console.log()
```

File: types/svgicons2svgfont/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "allowSyntheticDefaultImports": true
```
Comment (peterblazejewicz) [inline types/svgicons2svgfont/tsconfig.json:18]:
```suggestion
        "forceConsistentCasingInFileNames": true
```
It's usefull for consumer of the DT package, not here (you won't get this with defalt generated code)

File: types/svgicons2svgfont/svgicons2svgfont-tests.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import SVGIcons2SVGFont from 'svgicons2svgfont';
```
Comment (peterblazejewicz) [inline types/svgicons2svgfont/svgicons2svgfont-tests.ts:1]:
```suggestion
import SVGIcons2SVGFont = require('svgicons2svgfont');
```
CJS import (with TS touch):
https://github.com/nfroidure/svgicons2svgfont#in-your-scripts

**Review by peterblazejewicz — COMMENTED** [review]
So those `@default` are optional, but would be nice to have them,
adding those would improve using typtes

**Comment by casaper** [comment]
> So those `@default` are optional, but would be nice to have them,
> adding those would improve using typtes

@peterblazejewicz 

Thanks for your review. I committed all your suggestions. 👍

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@casaper thx!

**Pull request merged**